core/app/helpers/spree/taxons_helper.rb

```diff
@@ -4,16 +4,26 @@ module TaxonsHelper
     # that we can use configurations as well as make it easier for end users to override this determination.  One idea is
     # to show the most popular products for a particular taxon (that is an exercise left to the developer.)
     def taxon_preview(taxon, max=4)
-      products = taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").limit(max)
+      products = taxon.active_products.select(*selection_sql).limit(max)
       if (products.size < max)
         products_arel = Spree::Product.arel_table
         taxon.descendants.each do |taxon|
           to_get = max - products.length
-          products += taxon.active_products.select("DISTINCT (spree_products.id), spree_products.*, spree_products_taxons.position").where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
+          products += taxon.active_products.select(*selection_sql).where(products_arel[:id].not_in(products.map(&:id))).limit(to_get)
           break if products.size >= max
         end
       end
       products
     end
+
+    private
+
+    def selection_sql
+      [
+        "DISTINCT (spree_products.id)",
```

Comment:
@grzlus If we add the `ON` keyword, then we could definitely move this to Arel, but you're right, this won't be compatible with other DBs. I think I'll just make an if statement for database type. Its not nice, but it'll do :(